tests/models/test_solution_strategy.py

```diff
@@ -439,3 +442,29 @@ def test_check_convergence(
         nonlinear_increment, residual, np.zeros(1), nl_params
     )
     assert (converged, diverged) == expected
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        # Mass balance must be nonlinear, no matter with or without fractres.
+        dict(model_name="mass_balance", num_fracs=0, is_nonlinear=True),
+        dict(model_name="mass_balance", num_fracs=1, is_nonlinear=True),
+        # Momentum balance without fractures is linear.
+        dict(model_name="momentum_balance", num_fracs=0, is_nonlinear=False),
+        dict(model_name="momentum_balance", num_fracs=1, is_nonlinear=True),
+        dict(model_name="mass_and_energy_balance", num_fracs=0, is_nonlinear=True),
+        dict(model_name="poromechanics", num_fracs=0, is_nonlinear=True),
+        # There was a bug here once #1350.
+        dict(model_name="thermoporomechanics", num_fracs=0, is_nonlinear=True),
+        dict(model_name="thermoporomechanics", num_fracs=1, is_nonlinear=True),
+    ],
+)
+def test_linear_or_nonlinear_model(params: dict):
+    """Tests that the base models are properly tagged as linear or nonlinear."""
+    model_name: str = params["model_name"]
+    num_fracs: int = params["num_fracs"]
+    is_nonlinear: bool = params["is_nonlinear"]
```

Comment:
Please add meshing parameters to ensure this model has no fractures and is discretized with a Cartesian grid (this will substantially limit the cost of this test)

Reply:
I marked the model as Cartesian explicitly, it did not make any existing test fail.